src/middleware/__tests__/apiKeyChecker.unit.test.ts

```diff
@@ -0,0 +1,47 @@
+import fs from 'fs';
+import { AuthenticationError } from '../../errors';
+import { checkApiKey } from '../apiKeyChecker';
+import type { NextFunction, Request, Response } from 'express';
+
+describe('Authorization middleware', () => {
```

Comment:
Only because I heard someone else say it earlier: authentication here rather than authorization :nerd_face:

Reply:
(I will get this wrong a billion times plz correct me every time!)